Util/timing.py

```diff
@@ -0,0 +1,34 @@
+import time
+
+def get_fps_timer(desired_fps: float):
```

Comment:
could we inject a logging function directly in the timer? we could remove that from the engine and coach. Don't know if add a lot, but it could be interesting.

Reply:
like sleep = get_fps_timer(30, logger=logger.debug)

Reply:
Feel free to add one if you want. I also thought of transferring the fps logging logic to it, but refrained due to the possible usage of the various variables (for example, detecting low framerates) outside of the timing function.

I think it would be a better idea to use None as the default option for the argument though, because I think it would be a better default behavior not to expect the timer to log if used.

Reply:
Yes i meant to use None as default argument. the logger function passed as argument was the self logging.debug of the engine/coach

Reply:
Instead we could pass a generic callback that take the fps as argument. we then could do whatever we want.

Reply:
Added an on_missed_callback hook. A proper callback function/closure can then be used to show the FPS. If it is never or rarely called, then one can safely assume that the FPS is at the desired value.

Reply:
awesome